lib/domain.gi

```diff
@@ -232,12 +232,18 @@ InstallImmediateMethod( Enumerator,
 InstallMethod( AsList,
     "for a domain with stored domain generators",
     [ IsDomain and HasGeneratorsOfDomain ],
-    D -> DuplicateFreeList( GeneratorsOfDomain( D ) ) );
+    function( D )
+        if IsDuplicateFreeList( GeneratorsOfDomain( D ) ) then
+            return GeneratorsOfDomain( D );
+        else
+            return DuplicateFreeList( GeneratorsOfDomain( D ) );
+        fi;
+    end );
 
 InstallMethod( Enumerator,
     "for a domain with stored domain generators",
     [ IsDomain and HasGeneratorsOfDomain ],
-    D -> DuplicateFreeList( GeneratorsOfDomain( D ) ) );
+    D -> AsList(D) );
```

Comment:
Do we have any tests for either of these methods, which compare the before/after situation?

Reply:
I have checked and I didn't find any. Do you want me to add some?

I'd also be fine to remove the `AsList` here and instead duplicate the function from above.

Reply:
```suggestion
    AsList );
```
It's fine for me this way, but I'd slightly simplify the code, as suggested here.

Reply:
of course. thanks :+1:

Reply:
of course. thanks :+1: